app/actions/user.js

```diff
@@ -1,8 +1,79 @@
+import{
+  AsyncStorage
+}from 'react-native';
 import * as types from './types';
+const BreakoutApi= require("breakout-api-client");
 
 export function setUser(user){
   return{
     type: types.SET_USER,
     payload:user
   }
 }
+
+export function setUsername(username){
+  return {
+    type: types.SET_USERNAME,
+    payload:username
+  }
+}
+
+export function setLoggingIn(logging){
+  return {
+    type: types.LOGGING_IN,
+    payload:logging
+  }
+}
+
+export function login(username,password){
+  return (dispatch,getState) => {
+    console.log(username+" "+password);
+    let curState = getState()
+    let loggingIn = curState.isLoggingIn;
+    console.log(BreakoutApi);
+    dispatch(setLoggingIn(true));
+    sendLogin(username,password).then((resp)=>{
```

Comment:
Can we use async / await here? E.g. 
```js
export async function login(username, password) { 
    // ...
    const tokens = await sendLogin(username, password);
   // ...
}```

Reply:
I made the async login call a separate function so the login call itself stays clean and simple, but the async part can be moved into this function easily if wished

Reply:
Yeah I'd say just move it in here and save us the unnecessary `.then()` call, which should make things easier to read

Reply:
okay, done and in the next commit